### DISCUSSION TITLE
Using cancel_previous_submissions YAML code magic is not working

### DISCUSSION THREAD

**Discussion by vanvu08t2** [codemagic-ci-cd/codemagic-docs#2541] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

When using cancel_previous_submissions (set to true or false), the following error is shown:

`1 validation error for Workflow publishing -> app_store_connect -> cancel_previous_submissions App Store review submission is required for cancel_previous_submissions.`

This error just happened to me. It was still working fine 2 or 3 weeks ago

### Expected results

CD works fine when using cancel_previous_submissions

### Actual results

`1 validation error for Workflow publishing -> app_store_connect -> cancel_previous_submissions App Store review submission is required for cancel_previous_submissions.`
<img width="798" alt="Screenshot 2024-05-20 at 10 05 33" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/22784076/4228af26-0141-41fc-a2db-b8fc4a0b940a">

**Comment by dtrdic** [comment, marked as answer]
hi @vanvu08t2, we already solved this issue over intercom chat, but I'll share it here as well.
`cancel_previous_submissions` works only in conjunction with App Store submission. If `submit_to_app_store` is set to false (or is missing, which is equivalent to false), then there is nothing to cancel, and because of that validation error occurs.
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/9c932f3d-a8d8-475b-b508-d7531cf922a3)
